sklearn/metrics/classification.py

```diff
@@ -1348,6 +1348,66 @@ def recall_score(y_true, y_pred, labels=None, pos_label=1, average='binary',
     return r
 
 
+def balanced_accuracy_score(y_true, y_pred, sample_weight=None):
+    """Compute the balanced accuracy
+
+    The balanced accuracy is used in binary classification problems to deal
+    with imbalanced datasets. It is defined as the arithmetic mean of sensitivity
+    (true positive rate) and specificity (true negative rate), or the average
+    accuracy obtained on either class.
+
+    The best value is 1 and the worst value is 0.
+
+    Read more in the :ref:`User Guide <balanced_accuracy_score>`.
+
+    Parameters
+    ----------
+    y_true : 1d array-like
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like
+        Estimated targets as returned by a classifier.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    Returns
+    -------
+    balanced_accuracy : float.
+        The average of sensitivity and specificity
+
+    See also
+    --------
+    recall_score
```

Comment:
also related to `roc_auc_score`: for binary `y_true` and `y_pred`, `balanced_accuracy_score(y_true, y_pred) == roc_auc_score(y_true, y_pred)` (but in general, `roc_auc_score` takes as its second argument non-binary scores)